cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -854,6 +888,395 @@ class TestAzureFileSystem : public ::testing::Test {
+      // ASSERT_MOVE_OK(data.container_name, missing_container);
+      // AssertFileInfo(fs(),
+      //                ConcatAbstractPath(missing_container,
+      //                PreexistingData::kObjectName), FileType::File);
+    }
+    // Renaming to an empty container can work if the source is also empty
+    auto new_empty_container = PreexistingData::RandomContainerName(rng_);
+    auto new_empty_container_client = CreateContainer(new_empty_container);
+    ASSERT_MOVE_OK(empty_container, new_empty_container);
+  }
+
+  void TestMoveContainerToPath() {
+    auto data = SetUpPreexistingData();
+    ASSERT_MOVE("missing-container", data.ContainerPath("new-subdir"), ENOENT);
+    EXPECT_RAISES_WITH_MESSAGE_THAT(
+        Invalid,
+        HasDirMoveToSubdirMessage(data.container_name, data.ContainerPath("new-subdir")),
+        fs()->Move(data.container_name, data.ContainerPath("new-subdir")));
+    EXPECT_RAISES_WITH_MESSAGE_THAT(
+        NotImplemented,
+        HasCrossContainerNotImplementedMessage(data.container_name,
+                                               "a-container/new-subdir"),
+        fs()->Move(data.container_name, "a-container/new-subdir"));
+  }
+
+  void TestCreateContainerFromPath() {
+    auto data = SetUpPreexistingData();
+    auto missing_path = data.RandomDirectoryPath(rng_);
+    ASSERT_MOVE(missing_path, "new-container", ENOENT);
+    EXPECT_RAISES_WITH_MESSAGE_THAT(
+        Invalid,
+        ::testing::HasSubstr("Creating files at '/' is not possible, only directories."),
+        fs()->Move(data.ObjectPath(), "new-file"));
+    auto src_dir_path = data.RandomDirectoryPath(rng_);
+    ASSERT_OK(fs()->CreateDir(src_dir_path, false));
+    EXPECT_OK_AND_ASSIGN(auto src_dir_info, fs()->GetFileInfo(src_dir_path));
+    EXPECT_EQ(src_dir_info.type(), FileType::Directory);
+    EXPECT_RAISES_WITH_MESSAGE_THAT(
+        NotImplemented,
+        HasCrossContainerNotImplementedMessage(src_dir_path, "new-container"),
+        fs()->Move(src_dir_path, "new-container"));
+  }
+
+  void TestMovePaths() {
+    Status st;
+    auto data = SetUpPreexistingData();
+    // When source doesn't exist.
+    ASSERT_MOVE("missing-container/src-path", "a-container/dest-path", ENOENT);
+    auto missing_path1 = data.RandomDirectoryPath(rng_);
+    ASSERT_MOVE(missing_path1, "missing-container/path", ENOENT);
+
+    // But when source exists...
+    if (!WithHierarchicalNamespace()) {
+      // ...and containers are different, we get an error message telling cross-container
+      // moves are not implemented.
+      EXPECT_RAISES_WITH_MESSAGE_THAT(
+          NotImplemented,
+          HasCrossContainerNotImplementedMessage(data.ObjectPath(),
+                                                 "missing-container/path"),
+          fs()->Move(data.ObjectPath(), "missing-container/path"));
+      GTEST_SKIP()
+          << "The rest of TestMovePaths is not implemented for non-HNS scenarios";
```

Comment:
Can we just `return` here?
Should we use `GTEST_SKIP()` here? We have some tests for non-HNS case.

Reply:
I would prefer to more loudly communicate that MOST of the tests are not in fact running. The `return` is too subtle.